backend/core/ai/sql_executor.py

```
@@ -0,0 +1,320 @@
+                else:
+                    end = normalized.find('*/')
+                    normalized = '' if end == -1 else normalized[end + 2:]
+                normalized = normalized.lstrip()
+            # 공백 축약
+            normalized = ' '.join(normalized.split())
+            first_kw = (normalized.split(None, 1)[0] if normalized else '').upper()
+
+            # 즉시 차단: 위험/세션 영향/메타 명령
+            disallowed_any = {
+                'COPY', 'DO', 'CALL', 'LISTEN', 'NOTIFY', 'EXECUTE',
+                'PG_SLEEP', 'PG_TERMINATE_BACKEND', 'COPYTO', 'COPYFROM', 'COPY',
```

Ruff (0.13.1)

Sets should not contain duplicate item `'COPY'`

Remove duplicate item

(B033)